ui/shared/ad/CustomAdBanner.tsx

```diff
@@ -0,0 +1,48 @@
+import { Flex, chakra, Tooltip, Image } from '@chakra-ui/react';
+import React, { useState, useEffect } from 'react';
+
+import useCustomBanners from 'ui/shared/ad/useCustomBanners';
+
+const CustomAdBanner = ({ className }: { className?: string }) => {
+  const customBanners = useCustomBanners().data || [];
+  const [ currentBannerIndex, setCurrentBannerIndex ] = useState(0);
+  useEffect(() => {
+    if (customBanners.length === 0) {
+      return;
+    }
+    const timer = setInterval(() => {
+      setCurrentBannerIndex((prevIndex) => (prevIndex + 1) % customBanners.length);
+    }, 60000);
+
+    return () => {
+      clearInterval(timer);
+    };
+  }, [ customBanners.length ]);
+
+  if (customBanners.length === 0) {
+    return null;
+  }
+
+  const currentBanner = customBanners[currentBannerIndex];
+
+  return (
+    <>
+      <Flex className={ className } h="90px" display={{ base: 'none', lg: 'flex' }}>
+        <Tooltip label={ currentBanner.text } aria-label={ currentBanner.text }>
+          <a href={ currentBanner.url } target="_blank" rel="noopener noreferrer">
+            <Image src={ currentBanner.desktop } alt={ currentBanner.text } height="100%" width="auto" borderRadius="10px"/>
+          </a>
+        </Tooltip>
+      </Flex>
+      <Flex className={ className } h="90px" display={{ base: 'flex', lg: 'none' }}>
+        <Tooltip label={ currentBanner.text } aria-label={ currentBanner.text }>
+          <a href={ currentBanner.url } target="_blank" rel="noopener noreferrer">
+            <Image src={ currentBanner.mobile } alt={ currentBanner.text } height="100%" width="auto" borderRadius="10px"/>
+          </a>
+        </Tooltip>
+      </Flex>
```

Comment:
I think there is no need to render both desktop and mobile banners simultaneously. Can it be just an if-statement for the image source?

<img width="1468" alt="image" src="https://github.com/blockscout/frontend/assets/22130104/8bc12dd0-a12a-4186-aa15-19b52604e86e">

Reply:
I'll try to fix this.